queries/cdmq/add-run.js

```diff
@@ -37,10 +39,25 @@ function save_ver(ver) {
   }
 }
 
+// Read an xz file and decompress to string
+async function decompressXzFile(filename) {
+  // Create a readable stream from file
+  const fileStream = Readable.toWeb(fs.createReadStream(filename));
+
+  // Decompress using XzReadableStream
+  const decompressedResponse = new Response(new XzReadableStream(fileStream));
+
+  // Get the decompressed content as string
+  const text = await decompressedResponse.text();
+  return text;
+}
+
 async function readNdjsonXzToString(filePath) {
   return new Promise((resolve, reject) => {
     try {
-      const fileStream = fs.createReadStream(filePath);
+      const fileStream = fs.createReadStream(filePath, {
+        highWaterMark: 5 * 1024 * 1024
```

Comment:
magic number?

Reply:
Well, yes, pretty much.  It's a number that keeps the memory usage reasonable.